humanize-duration.js

```diff
@@ -499,6 +499,41 @@
   // The main function is just a wrapper around a default humanizer.
   var humanizeDuration = humanizer({})
 
+  // Compare a collection of languages against the specified type (language, or fallback)
+  function hasLanguage (collection, userLanguages) {
+    var value, i
+    if (!Array.isArray(collection)) {
+      return
+    }
+    // Allow an override, useful for switching between options.languages and languages
+    for (i = 0; i < collection.length; i++) {
+      value = collection[i]
+      // Check if languages has the array value
+      if (languages.hasOwnProperty(value)) {
+        return languages[value]
+      } else if (userLanguages.hasOwnProperty(value)) {
+        return userLanguages[value]
+      }
+    }
+  }
+  // Get the dictionary from the supplied language and possible fallback(s)
+  function getDictionary (options) {
```

Comment:
Rather than treating `options.language` and `options.fallbacks` separately, could we instead do something like:

```javascript
function getDictionary (options) {
  var languagesFromOptions = [options.language]

  if (options.hasOwnProperty('fallbacks')) {
    if (isArray(options.fallbacks) && options.length) {
      languagesFromOptions = languagesFromOptions.concat(options.fallbacks)
    } else {
      throw new Error('fallbacks must be an array with at least one element')
    }
  }

  for (var i = 0; i < languagesFromOptions.length; i++) {
    var languageToTry = languagesFromOptions[i]
    if (options.languages.hasOwnProperty(languageToTry)) {
      return options.languages[languageToTry]
    } else if (languages.hasOwnProperty(languageToTry)) {
      return languages[languageToTry]
    }
  }

  throw new Error('No language found.')
}

// We need to make sure we support browsers that don't have
// `Array.isArray`, so we define a fallback here.
var isArray = Array.isArray || function (arg) {
  return Object.prototype.toString.call(arg) === '[object Array]'
}
```